Sources/Commands/SwiftTool.swift

```diff
@@ -580,8 +584,83 @@ public class SwiftTool<Options: ToolOptions> {
         }
     }
 
-    /// Build a subset of products and targets using llbuild.
-    func build(plan: BuildPlan, subset: BuildSubset) throws {
+    func build(subset: BuildSubset) throws {
+        let buildParameters = try self.buildParameters()
+
+        let haveBuildManifestAndDescription =
+            localFileSystem.exists(buildParameters.llbuildManifest) &&
+            localFileSystem.exists(buildParameters.buildDescriptionPath)
+
+        // Run the build with the existing build description if we're asked to by-pass build planning.
+        if options.skipBuildPlanning && haveBuildManifestAndDescription {
+            let buildDescription = try BuildDescription.load(from: buildParameters.buildDescriptionPath)
+            return try build(buildDescription: buildDescription, subset: subset)
+        }
+
+        // Perform steps for build manifest caching if we can enabled it.
+        let hasEditedPackages = try getActiveWorkspace().managedDependencies.hasEditedPackages()
+        if options.enableBuildManifestCaching && haveBuildManifestAndDescription && !hasEditedPackages {
+            // Create the build system.
+            let buildSystem = try createBuildSystem()
+            buildSystemRef.buildSystem = buildSystem
+
+            // Build the package structure target which will re-generate the llbuild manifest, if necessary.
+            if !buildSystem.build(target: "PackageStructure") {
+                throw Diagnostics.fatalError
+            }
+
+            // Use the build description on disk to perform the build. We trust the above build to
+            // update the build description when needed.
+            let newBuildDescription = try BuildDescription.load(from: buildParameters.buildDescriptionPath)
```

Comment:
How big is the build description? Should we just put it inside the database?

Reply:
Currently, it just has a map of command name to target name for Swift targets. How would I put it in the db and how would I access it?

Reply:
Oh maybe we can’t do this yet because of the bad llbuild API; in general you would just serialize it to bytes. I think the problem is we only expose an ExternalCommand extension point